projects/dotcms-field-elements/src/components/dot-textfield/dot-textfield.tsx

```diff
@@ -8,42 +8,106 @@ import Fragment from 'stencil-fragment';
 })
 export class DotTextfieldComponent {
     @Prop() value: string;
+    @Prop() name: string;
     @Prop() regexcheck: string;
+    @Prop() regexcheckmessage: string;
     @Prop() readOnly: string;
     @Prop() label: string;
     @Prop() hint: string;
     @Prop() placeholder: string;
     @Prop() required: boolean;
-    @Event() onChange: EventEmitter;
+    @Prop() requiredmessage: string;
+    @Prop() disabled: boolean;
+    @Event() valueChanges: EventEmitter;
+    @Event() statusChanges: EventEmitter;
 
     @State() _value: string;
     @State() _error = false;
+    @State() _dotTouched = false;
+    _dotPristine = true;
+
     _label: string;
 
     // tslint:disable-next-line:cyclomatic-complexity
-    validate(value: string): boolean {
-        if (this.required && value.length === 0) {
-            return true;
+    isValid(): boolean {
+        if (this.required && this._value.length === 0) {
+            return false;
         }
 
         if (this.regexcheck) {
             const regex = new RegExp(this.regexcheck, 'ig');
-            return !regex.test(value);
+            return regex.test(this._value);
         }
 
-        return false;
+        return true;
+    }
+
+    setTouched(): void {
+        if (!this._dotTouched) {
+            this._dotTouched = true;
+            this.emitStatusChange();
+        }
+    }
```

Comment:
why this method just set touched to true?, it is weird a methos setXXX without parameters

Reply:
change to a better name and discussed with Freddy R.